narwhals/_arrow/series.py

```diff
@@ -70,6 +118,35 @@ def __mul__(self, other: Any) -> Self:
         other = validate_column_comparand(other)
         return self._from_native_series(pc.multiply(self._native_series, other))
 
+    def __truediv__(self, other: Any) -> Self:
+        pc = get_pyarrow_compute()
+        ser = self._native_series
+        other = validate_column_comparand(other)
+        return self._from_native_series(pc.divide(ser, other))
+
+    def __floordiv__(self, other: Any) -> Self:
+        pc = get_pyarrow_compute()
+        ser = self._native_series
+        other = validate_column_comparand(other)
+        return self._from_native_series(pc.floor(pc.divide(ser, other)))
+
+    def __pow__(self, other: Any) -> Self:
+        pc = get_pyarrow_compute()
+        ser = self._native_series
+        other = validate_column_comparand(other)
+        return self._from_native_series(pc.power(ser, other))
+
+    def __mod__(self, other: Any) -> Self:
+        pc = get_pyarrow_compute()
+        ser = self._native_series
+        other = validate_column_comparand(other)
+        res = pc.subtract(ser, pc.multiply(pc.divide(ser, other), other))
```

Comment:
As `__mod__` should work among integers, I am exploiting the fact that `pc.divide` will yield the integer division
(namely the equivalent of `s - s//other * other`)